scene/main/scene_tree.cpp

```diff
@@ -576,14 +576,13 @@ void SceneTree::finalize() {
 
 	initialized = false;
 
-	MainLoop::finalize();
-
 	if (root) {
 		root->_set_tree(nullptr);
 		root->_propagate_after_exit_tree();
 		memdelete(root); //delete root
 		root = nullptr;
 	}
+	MainLoop::finalize();
```

Comment:
I'd move it even at the end of this method as I think it should be done after finalizing all `SceneTree`-related stuff (and timers/tweens are `SceneTree`-specific).

Also while you're at it, you could move that `_flush_delete_queue();` from below (with the comment) at the end of the `if (root)` block above. As the comment states it's relevant only after destructing the `root` (for reference I'm the one who added it in #53571).

Reply:
> I'd move it even at the end of this method as I think it should be done after finalizing all SceneTree-related stuff (and timers/tweens are SceneTree-specific).

At first I did it like this, but it breaks construction-destruction order. because `MainLoop::finalize` calls gdscript method and SceneTree init-deinit order becomes:
1. SceneTree native init
2. SceneTree gdscript init (called from MainLoop::initialize)
3. SceneTree native finalize 
4. SceneTree gdscript finalize (called from MainLoop::finalize)

So I have decided to move it to this place, and not to redo script attaching for this class hierarchy. 

>Also while you're at it, you could move that _flush_delete_queue(); from below (with the comment) at the end of the if (root) block above

Ok.

Reply:
I guess it makes sense to leave it in here. In case someone would create timer/tween in script's `_finalize` then there will be no leak (contrary to if moved `MainLoop::finalize()` to the end of `SceneTree::finalize() like I've suggested`).

Reply:
Let's merge then?:)

Reply:
> Let's merge then?:)

Won't happen without approval from **reduz**, he's the maintainer of this area. I can tell you that it's already added to his personal project with PRs to review. But when he'll look at this I have no idea. :slightly_smiling_face: